### ISSUE TITLE
how should star-tuple and c_array arguments behave for interoperability?

### ISSUE THREAD

**Issue by mppf** [chapel-lang/chapel#15892]
Are tuples supported as an argument type for `extern` functions in the near term?

What does the `in` intent mean for tuples and c_array arguments? Note that a C argument like `void f(int[4])` always passes the fixed-sized array as a pointer rather than by value.

**Comment by bradcray** [comment]
I'd be fine with saying that c_arrays (or tuples) can't be passed by `in` intent to extern functions.  W.r.t. the varargs reservation you mention, are you imagining an `extern proc foo(in x...);` type of signature?  (I don't usually use `in` intent for printf in practice... but maybe with your recent "require intents for now" effort, you're saying it's the most natural way to write it?).

We were talking yesterday about whether a default intent for an `extern` routine should mean "the same as Chapel's default intent" or "the external lagnuage's default intent."  Maybe printf() suggests that the latter is the right choice?

**Comment by mppf** [comment]
> W.r.t. the varargs reservation you mention, are you imagining an `extern proc foo(in x...);` type of signature?  

Yes that was the concern. I think in practice, C functions that are varargs often expect each argument to be 1 pointer in size, but I don't know if that is required in all cases. (But our interoperability does not need to necessarily support all C varargs functions either).
